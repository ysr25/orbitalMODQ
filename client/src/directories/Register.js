import React, { Component } from "react";
import { Link } from "react-router-dom";
import settings from "../config/settings";
import UserForm from "../components/UserForm";

export default class Register extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      course: "",
      yearOfStudy: "matriculatingSoon", // default option
      username: "",
      password: "",

      status: null,
      isButtonDisabled: false,
    };
  }

  onChangeEmail = (data) => {
    this.setState({
      email: data,
      isButtonDisabled: false,
    });
  };

  onChangeCourse = (data) => {
    this.setState({
      course: data,
      isButtonDisabled: false,
    });
  };

  onChangeYearOfStudy = (data) => {
    this.setState({
      yearOfStudy: data,
      isButtonDisabled: false,
    });
  };

  onChangeUsername = (data) => {
    this.setState({
      username: data,
      isButtonDisabled: false,
    });
  };

  onChangePassword = (data) => {
    this.setState({
      password: data,
      isButtonDisabled: false,
    });
  };

  onSubmit = (e) => {
    e.preventDefault();

    const newUser = {
      email: this.state.email,
      course: this.state.course,
      yearOfStudy: this.state.yearOfStudy,
      username: this.state.username,
      password: this.state.password,
    };

    this.setState({
      status: "Submitting...",
      isButtonDisabled: true,
    });

    this.props
      .api("post", "/users/signup", newUser)
      .then((res) => {
        this.setState({ status: res.data.message });
        this.props.history.push("/");
      })
      .catch((err) => {
        this.setState({
          status: err.response.data.message,
          isButtonDisabled: true,
        });
      });
  };

  render() {
    return (
      <>
        <h3>Create An Account</h3>
        <p>
          Already have an account? <Link to="/users/login">Log in</Link>
        </p>
        <UserForm
          email={this.state.email}
          username={this.state.username}
          password={this.state.password}
          course={this.state.course}
          yearOfStudy={this.state.yearOfStudy}
          status={this.state.status}
          isButtonDisabled={this.state.isButtonDisabled}
          disabled={false}
          displayPassword={true}
          onChangeEmail={this.onChangeEmail}
          onChangeUsername={this.onChangeUsername}
          onChangePassword={this.onChangePassword}
          onChangeCourse={this.onChangeCourse}
          onChangeYearOfStudy={this.onChangeYearOfStudy}
          onSubmit={this.onSubmit}
        />
        <br />
        <a
          href={settings.serverUrl + "api/users/login/google"}
          className="btn btn-primary"
        >
          Sign up with Google
        </a>
      </>
    );
  }
}
