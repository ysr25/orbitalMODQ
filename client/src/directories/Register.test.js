import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Register from "./Register";

it("renders register page", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
