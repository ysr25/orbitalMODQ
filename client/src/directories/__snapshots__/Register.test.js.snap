// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders register page 1`] = `
Array [
  <h3>
    Create An Account
  </h3>,
  <p>
    Already have an account? 
    <a
      href="/users/login"
      onClick={[Function]}
    >
      Log in
    </a>
  </p>,
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="email"
      >
        Email
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          disabled={false}
          id="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
    </div>
    <br />
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="course"
      >
        Course
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="course"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <br />
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="yearOfStudy"
      >
        Year of Study
      </label>
      <div
        className="col"
      >
        <select
          className="form-control"
          id="yearOfStudy"
          onChange={[Function]}
          required={true}
          value="matriculatingSoon"
        >
          <option
            value="matriculatingSoon"
          >
            Matriculating Soon
          </option>
          <option
            value="undergraduate"
          >
            Undergraduate
          </option>
          <option
            value="masters"
          >
            Masters
          </option>
          <option
            value="doctorate"
          >
            Doctorate
          </option>
          <option
            value="others"
          >
            Others
          </option>
        </select>
      </div>
    </div>
    <br />
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="username"
      >
        Username
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          disabled={false}
          id="username"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <br />
    <div
      className="form-row"
    >
      <label
        className="form-label col-form-label col-sm-2"
        htmlFor="password"
        style={
          Object {
            "display": "block",
          }
        }
      >
        Password
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
          type="password"
          value=""
        />
      </div>
    </div>
    <br />
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
    <br />
  </form>,
  <br />,
  <a
    className="btn btn-primary"
    href="http://localhost:3001/api/users/login/google"
  >
    Sign up with Google
  </a>,
]
`;
